site-src/concepts/guidelines.md

```diff
@@ -1,4 +1,4 @@
-# Implementation guidelines
```

Comment:
I started this PR reviewing this page, but it turned out that these guidelines are actually for us, the project contributors and maintainers, not for implementers. So "Design guidelines" felt more appropriate.